base_classes/NXdata.nxdl.xml

```diff
@@ -85,124 +85,170 @@
 		.. index:: axes (attribute)
 		.. index:: coordinates
 
-		The :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields contain the axis coordinates associated with the data values.
-		The names of all :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields are listed in the
-		:ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute.
+		The :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields contain the axis coordinates associated with the signal values.
 
-		`Rank`
+		`Default axes`
 
-		:ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields are typically one-dimensional arrays, which annotate one of the dimensions.
+		The :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute provides the names of the :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;`
+		fields to be used as the default axis for each dimension of the :ref:`data &lt;/NXdata/DATA-field&gt;` fields.
+		As a result the length of the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute must be equal to the rank of the :ref:`data &lt;/NXdata/DATA-field&gt;`
+		fields. When a particular dimension has no default axis, the string “.” is used in that position.
 
-		An example of this would be
+		`Spanned dimensions`
+
+		The :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attributes describe the
+		:ref:`data &lt;/NXdata/DATA-field&gt;` dimensions spanned by the corresponding :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields.
+
+		When :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` is missing for a given :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field,
+		the position(s) of the :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field name in the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute
+		is(are) used. The shape of an :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field must be equal to the shape of the
```

Comment:
The values at those dimensions must match: `AXISNAME.shape[i] == SIGNAL.shape[AXISNAME_indices[i]]` for each `i` in `[0, AXISNAME.ndim)`

Reply:
This is worth pointing out, but it needs amending. Since the earliest days of the format, NeXus allowed axis dimensions to have a size of the corresponding data dimension + 1, because neutron facilities typically store time-of-flight bin boundaries, rather than centres.

Reply:
Yes that is what I mean with "matching shapes" as in [numpy.shape](https://numpy.org/doc/stable/reference/generated/numpy.shape.html) but I guess this is not commonly understood.

Reply:
I added your suggestion after this sentence:

The shape of an AXISNAME field must be equal to the shape of the DATA dimensions it spans. This means that `AXISNAME.shape[i] == DATA.shape[AXISNAME_indices[i]]` for each `i` in `[0, AXISNAME.ndim)`.

Reply:
Ray suggests that `== x` should be slacken to `== x or == x + 1`